src/forms/yup.js

```diff
@@ -19,6 +19,14 @@ import * as yup from 'yup';
 
 const ARRAY_INDEX_REGEX = /\[([^)]+)\]/;
 
+yup.addMethod(yup.string, 'selected', function () {
+  return this.test(
+    'selected',
+    params => ({ key: 'form.validation_select_field_required', params }),
+    value => Boolean(value)
```

Comment:
Suggestion: I don't think we need to define a new way to check if something is required, I think you should be able to use required() inside this and define the custom message, something like: 
```javascript
yup.addMethod(yup.string, 'selected', function () {
  return this.required(params => ({
    key: 'form.validation_select_field_required',
    params,
  }));
});
```